mautic/mautic
# Report email stats enhacement

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR change logic to generate email stats (Data Source: Email).

Before update the emails are listed with date_added limit  and numbers (sent, read, unsubscribed) are loaded global for all period. This a lot of users make confused.

Now
- If Hide date range is enabled - the dater range filter don't apply https://github.com/mautic/mautic/pull/7214/files#diff-5295a31f5d6a0e3c5ab1b7d4bc36f45dR373  . Thats what we expect, If I don't see date range forms, then date range cannot be apply to query.
- Data Source: Email - with date range now loaded emails active from date range and related stats (unsubscribed, ratio, read etc). That means If choose range last month, then I will see just emails and stats to that related month. 

With enabled **Hide date range** SQL use the global dat from **emails table** (sent_count, read_count ...)
With disable **Hide date range** SQL query create **inner JOIN** to other huge tables (page_hits, email_stats..) with date range limit. This prevent load the large table more times. 

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com/7214) or load to you local enviroment with stats
2. Jus retest both case
- If Hide date range is enabled/disabled
![image](https://user-images.githubusercontent.com/462477/52280292-dee7b280-295b-11e9-9622-23c76c103e95.png)
- If enabled - then try load custom date range and see If stats are related and what do you expect from that range
If disabled - see global stats to each email
Also try filters for both state

![image](https://user-images.githubusercontent.com/462477/52279816-b8754780-295a-11e9-84d4-a595c3bad580.png)


#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1. 
2.
